packages/edit-site/src/components/global-styles/use-global-styles-output.js

```diff
@@ -49,7 +49,11 @@ function compileStyleValue( uncompiledValue ) {
 	const VARIABLE_REFERENCE_PREFIX = 'var:';
 	const VARIABLE_PATH_SEPARATOR_TOKEN_ATTRIBUTE = '|';
 	const VARIABLE_PATH_SEPARATOR_TOKEN_STYLE = '--';
-	if ( uncompiledValue?.startsWith( VARIABLE_REFERENCE_PREFIX ) ) {
+
+	if (
+		typeof uncompiledValue === 'string' &&
+		uncompiledValue.startsWith( VARIABLE_REFERENCE_PREFIX )
+	) {
```

Comment:
Shorter alternative:

```suggestion
	if ( uncompiledValue?.startsWith?.( VARIABLE_REFERENCE_PREFIX ) ) {
```